gobblin-temporal/src/main/java/org/apache/gobblin/temporal/ddm/work/assistance/Help.java

```diff
@@ -0,0 +1,208 @@
+  public static final String AZKABAN_FLOW_EXEC_ID_KEY = "azkaban.flow.execid";
+  public static final String USER_TO_PROXY_KEY = "user.to.proxy";
+
+  // treat `JobState` as immutable and cache, for reuse among activities executed by the same worker
+  private static final transient Cache<Path, JobState> jobStateByPath = CacheBuilder.newBuilder().recordStats().build();
+  private static final transient AtomicInteger jobStateAccessCount = new AtomicInteger(0);
+
+  private Help() {}
+
+  public static String qualifyNamePerExec(String name, FileSystemJobStateful f, Config workerConfig) {
+    return name + "_" + calcPerExecQualifier(f, workerConfig);
+  }
+
+  public static String qualifyNamePerExec(String name, Config workerConfig) {
+    return name + "_" + calcPerExecQualifier(workerConfig);
+  }
+
+  public static String calcPerExecQualifier(FileSystemJobStateful f, Config workerConfig) {
+    Optional<String> optFlowExecId = Optional.empty();
+    try {
+      optFlowExecId = Optional.of(loadJobState(f).getProp(ConfigurationKeys.FLOW_EXECUTION_ID_KEY, null));
+    } catch (IOException e) {
+      log.warn("unable to loadJobState", e);
+    }
+    return optFlowExecId.map(x -> x + "_").orElse("") + calcPerExecQualifier(workerConfig);
+  }
+
+  public static String calcPerExecQualifier(Config workerConfig) {
+    String userToProxy = workerConfig.hasPath(USER_TO_PROXY_KEY)
+        ? workerConfig.getString(USER_TO_PROXY_KEY) : "";
+    String azFlowExecId = workerConfig.hasPath(AZKABAN_FLOW_EXEC_ID_KEY)
+        ? workerConfig.getString(AZKABAN_FLOW_EXEC_ID_KEY) : UUID.randomUUID().toString();
+    return userToProxy + "_" + azFlowExecId;
+  }
+
+  public static FileSystem loadFileSystem(FileSystemApt a) throws IOException {
+    // NOTE: `FileSystem.get` appears to implement caching, which should facilitate sharing among activities executing on the same worker
+    return loadFileSystemForUri(a.getFileSystemUri(), a.getFileSystemConfig());
+  }
+
+  public static FileSystem loadFileSystemForUri(URI fsUri, State fsConfig) throws IOException {
+    // TODO - determine whether this works... unclear whether it led to "FS closed", or that had another cause...
+    // return HadoopUtils.getFileSystem(fsUri, fsConfig);
+    Configuration conf = HadoopUtils.getConfFromState(fsConfig);
+    return FileSystem.get(fsUri, conf);
+  }
+
+  public static FileSystem loadFileSystemForce(FileSystemApt a) throws IOException {
+    return loadFileSystemForUriForce(a.getFileSystemUri(), a.getFileSystemConfig());
+  }
+
+  public static FileSystem loadFileSystemForUriForce(URI fsUri, State fsConfig) throws IOException {
+    // TODO - determine whether this works... unclear whether it led to "FS closed", or that had another cause...
+    // return HadoopUtils.getFileSystem(fsUri, fsConfig);
+    Configuration conf = HadoopUtils.getConfFromState(fsConfig);
+    conf.setBoolean("fs.hdfs.impl.disable.cache", true);
```

Comment:
Do we really want to disable the cache here and not utilize the Hadoop caching of the filesystems in `Filesystem.get`? Comments a bit unclear on the motivation around this

Reply:
this needs more testing.  I had difficulty finding repeatable success when caching was not disabled.  I'll improve the comment, but will leave it disabled for now, until we investigate further.